packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -161,30 +162,76 @@ export enum SObjectCategory {
   CUSTOM = 'CUSTOM'
 }
 
+type SubRequest = { method: string; url: string };
+type BatchRequest = { batchRequests: SubRequest[] };
+
+type SubResponse = { statusCode: number; result: SObject };
+
+type BatchResponse = { hasErrors: boolean; results: SubResponse[] };
+
 export class SObjectDescribe {
+  private accessToken: string;
+  private instanceUrl: string;
+  // TODO should get the proper version from ??
+  private readonly servicesPath: string = 'services/data';
+  private readonly targetVersion = '40.0';
+  private readonly versionPrefix = 'v' + this.targetVersion;
+  private readonly sobjectsPart: string = this.versionPrefix + '/sobjects';
+  private readonly batchPart: string = this.versionPrefix + '/composite/batch';
+
+  // get the token and url by calling the org - short term, should be able to get it from the sfdx project
+  private async setupConnection(projectPath: string, username?: string) {
+    if (!this.accessToken) {
+      let orgInfo: any;
+      const builder = new SfdxCommandBuilder().withArg('force:org:display');
+      if (username) {
+        builder.args.push('--targetusername', username);
+      }
+      const command = builder.withJson().build();
+      const execution = new CliCommandExecutor(command, {
+        cwd: projectPath
+      }).execute();
+      const cmdOutput = new CommandOutput();
+      const result = await cmdOutput.getCmdResult(execution);
+      orgInfo = JSON.parse(result).result;
+      this.accessToken = orgInfo.accessToken;
+      this.instanceUrl = orgInfo.instanceUrl;
+    }
+  }
   public async describeSObject(
```

Comment:
Newline between methods.

Reply:
wonder why lint didnt complain - should it have?  - no problem, will fix...

Reply:
It can't :-(

Or at least, I don't know how to enforce it with either tslint or prettier.